vertx-web/src/main/java/io/vertx/ext/web/impl/HandlersCollection.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2023 Red Hat, Inc.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ * The Eclipse Public License is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * The Apache License v2.0 is available at
+ * http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.ext.web.impl;
+
+import io.vertx.core.Handler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+class HandlersCollection<E> {
+
+  private final List<Handler<E>> list;
+
+  HandlersCollection() {
```

Comment:
for debugging maybe worth adding a proper size counter with the actual number of element in?

It could be used to trigger an actual resize of the list in case all the elements are removed too (not sure is useful so take it with grain of salt)

Reply:
> for debugging maybe worth adding a proper size counter with the actual number of element in?

In most, if not all cases, the list size will be very small. It shouldn't be difficult to inspect its content with the debugger to find out how many array elements are not null.

> It could be used to trigger an actual resize of the list in case all the elements are removed too (not sure is useful so take it with grain of salt)

I don't think this is useful. The `RoutingContext` is not a long-lived object. It goes away with the request and response objects.